### PR TITLE
fixing /project endpoint and refactoring

### PR DESCRIPTION
**Pull request by cubitouch**
Refactoring and solving the original problem according to this PR : https://github.com/OmniSharp/omnisharp-roslyn/pull/67

### CODE REVIEW

**Comment by jchannon** [comment]
:thumbsup: Any chance of some tests before @nosami sends the boys around

**Comment by cubitouch** [comment]
Not sure it was a question, if you need me to help test something or anything, just ask :)

**Comment by davidfowl** [comment]
You're using tabs, please fix them

**Comment by cubitouch** [comment]
How do I configure that in the CTP5 ?

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs [hunk 1]
```diff
@@ -2,29 +2,60 @@
...
+			};
+		}
+
+		[HttpPost("/project")]
+		[HttpGet("/project")]
+		public ProjectInformationResponse CurrentProject(Request request)
+		{
+			var document = _workspace.GetDocument(request.FileName);
+
+			ProjectFileInfo msBuildContextProject = _msbuildContext.GetProject(document.Project.FilePath);
```
Comment (davidfowl) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs]: Use var

File: src/OmniSharp/MSBuild/MSBuildContext.cs [hunk 2]
```diff
@@ -10,5 +10,16 @@ public class MSBuildContext
         public Dictionary<Guid, ProjectId> ProjectGuidToWorkspaceMapping { get; } = new Dictionary<Guid, ProjectId>();
         public Dictionary<string, ProjectFileInfo> Projects { get; } = new Dictionary<string, ProjectFileInfo>();
         public string SolutionPath { get; set; }
-    }
+		
+		public ProjectFileInfo GetProject(string path)
+		{
+			ProjectFileInfo p;
```
Comment (davidfowl) [inline src/OmniSharp/MSBuild/MSBuildContext.cs]: project instead of p

File: src/OmniSharp/Models/AspNet5Project.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	{
+		public string Path { get; set; }
+		public string Name { get; set; }
+		public IDictionary<string, string> Commands { get; set; }
+		public IList<string> Configurations { get; set; }
+		public IList<string> ProjectSearchPaths { get; set; }
+		public IList<string> Frameworks { get; set; }
+		public string GlobalJsonPath { get; set; }
+
+		public AspNet5Project(Project p)
```
Comment (davidfowl) [inline src/OmniSharp/Models/AspNet5Project.cs]: project instead of p

File: src/OmniSharp/Properties/debugSettings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (davidfowl) [inline src/OmniSharp/Properties/debugSettings.json]: Remove this from the commit and add it to the ignore file

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs [hunk 5]
```diff
@@ -26,5 +29,33 @@ public WorkspaceInformationResponse ProjectInformation()
...
             };
         }
+
+        [HttpPost("/project")]
+        [HttpGet("/project")]
+        public ProjectInformationResponse CurrentProject(Request request)
+        {
+            var document = _workspace.GetDocument(request.FileName);
+
+            ProjectFileInfo msBuildContextProject = _msbuildContext.GetProject(document.Project.FilePath);
```
Comment (davidfowl) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs]: Use var

File: src/OmniSharp/MSBuild/MSBuildContext.cs [hunk 6]
```diff
@@ -10,5 +10,16 @@ public class MSBuildContext
         public Dictionary<Guid, ProjectId> ProjectGuidToWorkspaceMapping { get; } = new Dictionary<Guid, ProjectId>();
         public Dictionary<string, ProjectFileInfo> Projects { get; } = new Dictionary<string, ProjectFileInfo>();
         public string SolutionPath { get; set; }
+
+        public ProjectFileInfo GetProject(string path)
+        {
+            ProjectFileInfo p;
```
Comment (davidfowl) [inline src/OmniSharp/MSBuild/MSBuildContext.cs]: projectFileInfo instead of p

File: src/OmniSharp/Models/AspNet5Project.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public string Path { get; set; }
+        public string Name { get; set; }
+        public IDictionary<string, string> Commands { get; set; }
+        public IList<string> Configurations { get; set; }
+        public IList<string> ProjectSearchPaths { get; set; }
+        public IList<string> Frameworks { get; set; }
+        public string GlobalJsonPath { get; set; }
+
+        public AspNet5Project(Project p)
```
Comment (davidfowl) [inline src/OmniSharp/Models/AspNet5Project.cs]: project instead of p

File: src/OmniSharp/Models/CurrentProjectResponse.cs [hunk 8]
```diff
@@ -1,9 +1,9 @@
 namespace OmniSharp.Models
 {
     using System.Collections.Generic;
-    
+
     public class CurrentProjectResponse
```
Comment (davidfowl) [inline src/OmniSharp/Models/CurrentProjectResponse.cs]: This file can be deleted

**Comment by jchannon** [comment]
Looks good. We had s discussion the other day that all new PRs would have unit/acceptance tests to show what new features are doing.

**Comment by jchannon** [comment]
#70 was the issue

**Comment by davidfowl** [comment]
@cubitouch add some tests

**Comment by cubitouch** [comment]
Just did, not sure if that's ok ?

**Comment by nosami** [comment]
Please rebase.

**Comment by cubitouch** [comment]
I close this one as we continued on an other: https://github.com/OmniSharp/omnisharp-roslyn/pull/82

**Pull request closed without merging**